source/element_parameters.md

```diff
@@ -0,0 +1,225 @@
+The default is `ENTRANCE_END`.
+
+### aperture_shape
+
+```{figure} figures/apertures.svg
+:width: 90%
+:name: f:aperture
+
+A) RECTANGULAR and ELLIPTICAL apertures. As drawn, `x_limit[1]` and `y_limit[1]` are 
+negative and `x_limit[2]` and `y_limit[2]` are positive. B) The `wall2d` aperture is defined
+by a set of vertices.
+```
+
+The `aperture_shape` parameter selects the shape of the aperture. Possible values are:
+```{code} yaml
+RECTANGULAR   # Rectangular shape.
+ELLIPTICAL    # Elliptical shape.
+WALL2D        # Shape defined by set of vertices.
+CUSTOM_SHAPE  # Shape defined outside of the lattice standard.
+```
+
+### x_limit and y_limit
+
+For `RECTANGULAR` and `ELLIPTICAL` shapes the `x_limit` and `y_limit` parameters are
+used to calculate the aperture as shown in {numref}`f:aperture`A. 
+For an `ELLIPTICAL` aperture, a particle with position {math}`(x, y)` is outside of the aperture if any 
+one of the following four conditions is true:
+```{code}
+  1) x < 0 and y < 0 and (x/x_limit[1])^2 + (y/y_limit[1])^2 > 1 
+  2) x < 0 and y > 0 and (x/x_limit[1])^2 + (y/y_limit[2])^2 > 1
+  3) x > 0 and y < 0 and (x/x_limit[2])^2 + (y/y_limit[1])^2 > 1
+  4) x > 0 and y > 0 and (x/x_limit[2])^2 + (y/y_limit[2])^2 > 1
+```
+For a `RECTANGULAR` aperture the corresponding four conditions are:
+```{code}
+  1) x < x_limit[1]
+  2) x > x_limit[2]
+  3) y < y_limit[1]
+  4) y > y_limit[2]
+```
+
+Default values for the limits are `[-Inf, Inf]` for both `x_limit` and `y_limit`.
+
+### misalignment_moves_aperture
+
+The `misalignment_moves_aperture` parameter determines whether misaligning an element 
+affects the placement of the aperture. The default is `False`. 
+A common case where `misalignment_moves_aperture` would be `False` is when a beam pipe,
+which incorporates the aperture, is not physically touching the surrounding magnet element. 
+When tracking a particle, assuming that there are only apertures at the element ends, 
+the order of computation with `misalignment_moves_aperture` set to `False` could be
+```{code} yaml
+  1) Start at upstream end of element
+  2) Check upstream aperture if there is one.
+  3) Convert from branch coordinates to body coordinates.
+  4) Track through the element body.
+  5) Convert from body coordinates to branch coordinates.
+  6) Check downstream aperture if there is one.
+  7) End at downstream end of element.
+```
+With `misalignment_moves_aperture` set to `True`, the computation order could be
+```{code} YAML
+  1) Start at upstream end of element
+  2) Convert from branch coordinates to body coordinates.
+  3) Check upstream aperture if there is one.
+  4) Track through the element body.
+  5) Check downstream aperture if there is one.
+  6) Convert from body coordinates to branch coordinates.
+  7) End at downstream end of element.
+```
+
+### material
+
+The `material` parameter sets the material of the aperture. 
+Using chemical formulas like `Cu` and `Fe` are the most portable.
+
+### wall2d
+
```

Comment:
There could be a more descriptive name also for `wall2d`, such as `user_defined`, `aperture_user_defined`, or something similar.  Some name that reflects the flexibility to construct a very general aperture boundary.

Reply:
One possibility is `vertex`. `wall2d` is in reference to the fact that this description can be extended to `wall3d` which defines a 3D wall using an array of `wall2d` at different s-positions.

Reply:
Also there needs to be a distinction between `wall2d` and a custom program dependent defined aperture that is not described by the standard

Reply:
What about `vertex_defined`?

Reply:
Or how about `vertexes`?

Reply:
`vertices`?

Reply:
OK `vertices` it is.

Reply:
Done.